### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by elad237**
**the following steps were used for implementing the Decorator design pattern for this case.**
Interface
Create a class Nameable.
Implement a method called correct_name that will raise a NotImplementedError.
Turn your Person class to Nameable
Make sure that your Person class inherits from Nameable
Make sure that this class has a method correct_name implemented. It should simply return the name attribute.
Prepare base Decorator
Make sure that it inherits from Nameable.
In the constructor assign a nameable object from params to an instance variable.
Implement the correct_name method that returns the result of the correct_name method of the @nameable.
Prepare CapitalizeDecorator and TrimmerDecorator
For the CapitalizeDecorator:
Create a class that inherits from the base Decorator class.
Implement a method correct_name that capitalizes the output of @nameable.correct_name.
For the TrimmerDecorator:
Create a class that inherits from the base Decorator class.
Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

**Review by Felix45 — CHANGES_REQUESTED** [review]
Hi @elad237, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights :heavy_check_mark: 
- [x] `Namable` class with a method 'correct_name' raising an error is set up correctly :white_check_mark: 
- [x] Decorator classes are created correctly :white_check_mark: 
- [x]  Pull request has a descriptive title and summary :white_check_mark: 
- [x] Application works as expected :white_check_mark:  

  Screenshot from 2023-07-01 20-55-42

## Required Changes ♻️
- [ Required ] I think it would be better if you fix the License link on your readme file, currently, when I click on the link it leads to a `404` page not found error. I would like to suggest that you refactor the link as shown below because I've seen the name of the file is misspelled.

```ruby
  This project is [MIT](./LICENSE) licensed
```
## Error
Screenshot from 2023-07-01 20-57-25

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 1]
```diff
@@ -99,8 +99,8 @@ To run the project, execute the following command:
 
 ## 🔭 Future Features <a name="future-features"></a>
 
-- **Project 1: add Person, Student, and Teacher classes.**
-- **Project 2: use the "Decorator" design pattern.**
+- [x] Project 1: add Person, Student, and Teacher classes.
+- [x] Project 2: use the "Decorator" design pattern.
 - **Project 3: set up associations.**
 - **Project 4: add basic UI.**
 - **Project 5: refactor your code.**
```
Comment (Munsa1) [inline README.md:106]: - [ ] As requested by the previous reviewer, kindly fix the broken link to your `MIT` license agreement
image

It doesn't go to the license when a user clicks on it
image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @elad237 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
Passing Linters :heavy_check_mark: 
Correct GitHub Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by victorokoroji — APPROVED** [review]
Hi @elad237 , 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- Requested change implemented ✅ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**